File: Models/Api/Auth/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MemoLens.Models.Api.Auth;

public class RegisterRequest
{
    [StringLength(100, ErrorMessage = "Tên hiển thị không được dài quá 100 ký tự.")]
    public string? DisplayName { get; set; }

    [Required(ErrorMessage = "Email là bắt buộc.")]
    [EmailAddress(ErrorMessage = "Email không đúng định dạng.")]
    [StringLength(256, ErrorMessage = "Email không được dài quá 256 ký tự.")]
    public string Email { get; set; } = string.Empty;

    [Required(ErrorMessage = "Mật khẩu là bắt buộc.")]
    [StringLength(100, MinimumLength = 8, ErrorMessage = "Mật khẩu phải có từ 8 đến 100 ký tự.")]
    public string Password { get; set; } = string.Empty;

    [Required(ErrorMessage = "Vui lòng xác nhận mật khẩu.")]
    [Compare(nameof(Password), ErrorMessage = "Mật khẩu xác nhận không khớp.")]
    public string ConfirmPassword { get; set; } = string.Empty;
}

File: mobile/memolens_app/lib/features/authentication/data/models/register_request.dart
class RegisterRequest {
  const RegisterRequest({
    required this.email,
    required this.password,
    required this.confirmPassword,
    this.displayName,
  });

  final String? displayName;
  final String email;
  final String password;
  final String confirmPassword;

  Map<String, dynamic> toJson() => {
    'displayName': displayName?.trim(),
    'email': email.trim(),
    'password': password,
    'confirmPassword': confirmPassword,
  };
}
